sdk/appcenter-analytics/src/main/java/com/microsoft/appcenter/analytics/channel/SessionTracker.java

```diff
@@ -123,33 +131,43 @@ public void onPreparingLog(@NonNull Log log, @NonNull String groupName) {
     private void sendStartSessionIfNeeded() {
         if (mSid == null || hasSessionTimedOut()) {
 
-            /* New session: generate a new identifier. */
-            mSid = UUID.randomUUID();
-
-            /* Update session storage. */
-            SessionContext.getInstance().addSession(mSid);
-
             /*
              * Record queued time for the session log itself to avoid double log if resuming
              * from background after timeout and sending a log at same time we resume like a page.
              */
             mLastQueuedLogTime = SystemClock.elapsedRealtime();
 
-            /* Enqueue a start session log. */
-            StartSessionLog startSessionLog = new StartSessionLog();
-            startSessionLog.setSid(mSid);
-            mChannel.enqueue(startSessionLog, mGroupName, Flags.DEFAULTS);
+            /* Generate session id and send start session log. */
+            sendStartSession();
         }
     }
 
+    /**
+     * Generate session id and send start session log.
+     */
+    private void sendStartSession() {
+        mSid = UUID.randomUUID();
+
+        /* Update session storage. */
+        SessionContext.getInstance().addSession(mSid);
+
+        /* Enqueue a start session log. */
+        StartSessionLog startSessionLog = new StartSessionLog();
+        startSessionLog.setSid(mSid);
+        mChannel.enqueue(startSessionLog, mGroupName, Flags.DEFAULTS);
+    }
+
     /**
      * Call this whenever an activity is resumed to update session tracker state.
      */
     @WorkerThread
     public void onActivityResumed() {
+        if (isManualSessionTrackerEnabled) {
+            AppCenterLog.debug(Analytics.LOG_TAG, "Manual session tracker is enabled. Skip tracking a session status request after resumed activity.");
```

Comment:
I think we either don't need these logs in lifecycle events, or at lest may be make it verbose?